tests/e2e_application_aware_resource_quota_test.go

```diff
@@ -1388,6 +1388,35 @@ var _ = Describe("ApplicationAwareResourceQuota", func() {
 		Expect(nodeName).ToNot(BeNil())
 		Expect(*nodeName).ToNot(BeEmpty())
 
+		By("Adding a taint to the node")
+		node, err := f.K8sClient.CoreV1().Nodes().Get(context.Background(), *nodeName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		const taintKey = "testTaint"
+		node.Spec.Taints = append(node.Spec.Taints, v1.Taint{Key: taintKey, Value: "testValue", Effect: v1.TaintEffectNoSchedule})
+		node, err = f.K8sClient.CoreV1().Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		defer func() {
+			By("Removing a taint from the node")
+			node, err := f.K8sClient.CoreV1().Nodes().Get(context.Background(), *nodeName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			taintIdx := -1
+			for i, taint := range node.Spec.Taints {
+				if taint.Key == taintKey {
+					taintIdx = i
+				}
+			}
+
+			Expect(taintIdx).To(BeNumerically(">", 0), "test taint is not found")
+			// remove taint
+			node.Spec.Taints = append(node.Spec.Taints[:taintIdx], node.Spec.Taints[taintIdx+1:]...)
```

Comment:
can we maybe store the original node Taints in `BeforeEach` and restore it in `AfterEach` instead

Reply:
I thought about it as well, but it means I'll be creating a `Context()` only for a single test. I'm not sure if it's better or worse. WDYT?

Reply:
I mean, the Context would most probably have 1 test in the future too, since this case is very specific

Reply:
I prefer `Context()` even if it a single test

Reply:
done